src/bytes.rs

```diff
@@ -1116,11 +1118,10 @@ unsafe fn shared_to_vec_impl(shared: *mut Shared, ptr: *const u8, len: usize) ->
         .compare_exchange(1, 0, Ordering::AcqRel, Ordering::Relaxed)
         .is_ok()
     {
-        let buf = (*shared).buf;
-        let cap = (*shared).cap;
-
-        // Deallocate Shared
-        drop(Box::from_raw(shared as *mut mem::ManuallyDrop<Shared>));
+        // Deallocate the `Shared` instance without running its destructor.
+        let shared = ManuallyDrop::new(*Box::from_raw(shared));
```

Comment:
This is a big confusing. Please move the `Box::from_raw(shared)` to its own line.

Reply:
Sure thing! I agree that using `*` is confusing here. I'd prefer to use something like [`Box::into_inner`](https://doc.rust-lang.org/std/boxed/struct.Box.html#method.into_inner) if it was available. Regardless, I split this onto a separate line in b0aacc99e5a0e451ac6b8ffe1c27887dbb93d1c8.